### PR TITLE
Add optional install options with pip 

### PR DESCRIPTION
**Pull request by SarahG-579462**
## Types of changes

This PR allows users to install the optional dependencies found in `requirements.txt` with `pip install POT[all]`, or specific requirements with `pip install POT[backend-pytorch]` for example.

## Motivation and context / Related issue

This clarifies that `requirements.txt` are optional dependencies, and finds which modules are optional for specific parts of the code and adds them as an installation option.

The specific install options are: `backend-jax, backend-tf, backend-torch, cvxopt, dr, gnn, all`. 

Using `all` installs all modules previously available in requirements.txt (which is renamed requirements_opt.txt)

This solves the misunderstanding I had in #622, by clarifying that `cvxopt` is an optional dependency.

## How has this been tested (if it applies)

- running `pip install .[subparts]` to ensure that the requested modules are installed

## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [ ] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -71,6 +74,17 @@
     data_files=[],
     setup_requires=["oldest-supported-numpy", "cython>=0.23"],
     install_requires=["numpy>=1.16", "scipy>=1.6"],
+    extras_require={
+        'backend-numpy': [], # in requirements.
+        'backend-jax': ['jax<=0.4.24', 'jaxlib<=0.4.24'],
+        'backend-cupy': [], # should be installed with conda, not pip, or figure out what CUDA version above.
+        'backend-tf': ['tensorflow'],
+        'backend-torch': ['torch_geometric'],
```
Comment (rflamary) [inline setup.py:82]: this shouls be torch no?

Comment (rflamary) [inline setup.py:82]:
```suggestion
        'backend-torch': ['torch'],
```

**Review by rflamary — COMMENTED** [review]
This is nice.

I would like tu name the re quirement file requirements_all.txt instead of 
opt. Also It shoudo be used/upadted in the github actions

**Comment by rflamary** [comment]
Hello @SarahG-579462 ,

Thanks again for your contribution.

We plan on doing a release is a few days. It would be nice to have merged this PR that will allow better installation. Could you have  aquicl look at my comment and merge the conflcts with current setup.py?

**Comment by rflamary** [comment]
thanks @SarahG-579462 , we still need to change the requirement file name to `requirements_all.txt` and change the file name in all tests and doc build. After that all tests should pass and we can merge

**Comment by SarahG-579462** [comment]
> thanks @SarahG-579462 , we still need to change the requirement file name to `requirements_all.txt` and change the file name in all tests and doc build. After that all tests should pass and we can merge

Great, thanks for your help !

**Pull request merged**